### PR TITLE
[IN-820] Making AttributeStatement optional.

### PR DESCRIPTION
**Pull request by ddombrow**
Attributes will now be optional and set to empty array.  For CORT this will be accompanied by a PR to authenticator to attempt to pull email from whatever field they are sending it in.

### CODE REVIEW

File: lib/response-handling.js [hunk 1]
```diff
@@ -191,21 +191,30 @@ function checkStatus(doc, idp) {
...
+
+			return {
+				name: attrName,
+				friendlyName: friendlyName,
+				values: attrVals
+			};
+		});
+	}
+	else { 
+		//attribute statement is not required, try falling back to nameId
```
Comment (brownstein) [inline lib/response-handling.js]: Remove the console.log and error comments

File: lib/response-construction.js [hunk 2]
```diff
@@ -162,33 +162,35 @@ function constructAssertion(sp, idp, inResponseTo, nameID, attributes, issueInst
...
-			"saml:Attribute": [{
-				"@Name": attributeName,
-				"@NameFormat": protocol.ATTRNAMEFORMAT.BASIC
-			}].concat(attributeValues.map(value => {
-				return {"saml:AttributeValue": {
-					"@xsi:type": "xs:string",
-					"#text": value
-				}};
-			}))
+	if (attributes) {
```
Comment (brownstein) [inline lib/response-construction.js]: 👍  for modifying the IDP response to handle this variability

**Comment by ddombrow** [comment]
@brownstein updated.

**Comment by brownstein** [comment]
@ddombrow 💯

**Pull request merged**